doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,806 @@
+
+### 2.3. Detect and Debug
+* Health
+    * SmartSwitch DPUs should store their health data in their local StateDB 
+    * DPUs should support a CLI to display the health data “show chassis health-events”
+    * The host should be able to access this data using a redis call or an api
+* Alarm and Syslog
+    * Raise alarms when the temperature thresholds exceed, fans run slow or not present or faulty
+    * Drive LEDs accordingly
+    * Provide LED status indicators for DPU boards
+    * Trigger syslog
+* Console
+    * Provide console access to the DPUs through the Host CPU from the front panel management port
+    * The modular chassis console utility will be extended to access DPUs in place of LCs
+### 2.3. RMA
+* The DPUs should be displayed as part of inventory
+* Extend the CLI “show platform inventory” to display the DPUs
+* The system should be powered down for replacement of dpu-card
+
+## 3.	SmartSwitch PMON Design
+SmartSwitch PMON block diagram
+<p align="center"><img src="./images/pmon-blk-dgm.svg"></p>
+
+### 3.1. Platform monitoring and management
+* SmartSwitch design Extends the existing chassis_base and module_base as described below.
+* Extend MODULE_TYPE in ModuleBase class with MODULE_TYPE_DPU and MODULE_TYPE_SWITCH to support SmartSwitch
+
+#### 3.1.1 ChassisBase class API enhancements
+get_supervisor_slot(self):
+
+get_my_slot(self):
+```
+    Retrieves the slot number
+    Returns:
+      0 : on switch
+      1 : on DPU1
+      2 : on DPU2 and so on
+```
+
+is_modular_chassis(self):
+```
+    Retrieves whether the sonic instance is part of modular chassis. Smartswitch
+    chassis is fixed, the modular chassis class is extended to support DPUs
+
+    Returns:
+      False
+```
+
+get_num_modules(self):
+```
+    Retrieves the number of modules available on this chassis including DPUs
+
+    Returns:
+        An integer, the number of modules available on this chassis
+```
+
+get_all_modules(self):
+```
+    Retrieves all modules available on this chassis including DPUs
+
+    Returns:
+        A list of objects derived from ModuleBase representing all modules
+        available on this chassis
+```
+
+get_module(self, index):
+```
+    Retrieves module represented by index <index> switch:0, DPU1:1 and so on
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        An object derived from ModuleBase representing the specified module
+```
+
+get_module_index(self, module_name):
+```
+    Retrieves module index from the module name
+
+    Args:
+        module_name: A string, Ex. SWITCH, DPU1, DPU2 ... DPUX
+
+    Returns:
+        An integer, the index of the ModuleBase object in the module_list
+```
+#### 3.1.2 ChassisBase class new APIs
+is_smartswitch(self):
+```
+    Retrieves whether the sonic instance is part of smartswitch
+
+    Returns:
+      True
+```
+
+get_module_dpu_port(self, index):
```

Comment:
There are at least two ports connected to the DPU. One midplane (PCI) interface and one or more data plane interfaces. If I understand correctly, this API should return the mapping from the data plane port on the NPU to the data plane port on the DPU, e.g.: {"Ethernet224": "Ethernet0"}